haskell/nix.bzl

```diff
@@ -192,9 +191,9 @@ def haskell_nixpkgs_packageset(name, base_attribute_path, repositories = {}, **k
       `@hackage//:{your-package-name}`
     """
 
-    repositories = {
-        "bazel_haskell_wrapper": "@io_tweag_rules_haskell//haskell:nix/default.nix",
-    } + repositories
+    repositories = dict(repositories)
+    repositories.setdefault("bazel_haskell_wrapper", "@io_tweag_rules_haskell//haskell:nix/default.nix")
```

Comment:
I had never thought about this usage of `setdefault` and I find it rather difficult to read. Why not:

```
{ "bazel_haskell_wrapper": "@io_tweag_rules_haskell//haskell:nix/default.nix"}.update(repositories)
￼```

Reply:
`dict.update` doesn't return any value.
I've updated the code to use the skylib function `dicts.add`

Reply:
Indeed about `dict.update`, I wrote this too quickly ;) Good solution with `dicts.add`.